src/main/kotlin/step2/Calculator.kt

```diff
@@ -3,21 +3,14 @@ package step2
 class Calculator(private val expressionParser: ExpressionParser) {
 
     fun calculate(expression: String): Int {
-        expressionParser.parse(expression)
-        val operands = expressionParser.getOperandsInOrder()
-        val operators = expressionParser.getOperatorInOrder()
-
-        return calculate(operands, operators)
+        val expression = expressionParser.parse(expression)
```

Comment:
함수 인자와 지역변수 이름이 같아 shadowing이 발생하고 있네요🙂
`Expression`이 데이터 클래스인 점을 고려하면 다음과 같이 `Destructuring`을 적용해볼 수도 있을 것 같아요.
```suggestion
        val (operands, operators) = expressionParser.parse(expression)
```

Reply:
오 이런 기능이 있군요